configure

```diff
@@ -0,0 +1,404 @@
+: "${MANDIR:="/share/man/"}"
+: "${NO_SANITIZE:="auto"}"
+: "${SHUTDOWN_PREFIX:=""}"
+: "${CXXFLAGS_EXTRA:=""}"
+: "${TEST_CXXFLAGS_EXTRA:=""}"
+: "${CPPFLAGS:=""}"
+: "${LDFLAGS_EXTRA:=""}"
+: "${TEST_LDFLAGS_EXTRA:=""}"
+if [ "$PLATFORM" = "Linux" ]; then
+    : "${BUILD_SHUTDOWN:="yes"}"
+    : "${SUPPORT_CGROUPS:="1"}"
+    : "${SYSCONTROLSOCKET:="/run/dinitctl"}"
+else
+    : "${BUILD_SHUTDOWN:="no"}"
+    : "${SUPPORT_CGROUPS:="0"}"
+    : "${SYSCONTROLSOCKET:="/var/run/dinitctl"}"
+fi
+
+## Resolve final paths
+SBINDIR="$EPREFIX/$SBINDIR"
+MANDIR="$PREFIX/$MANDIR"
+
+## Finalize $CXXFLAGS, $TEST_CXXFLAGS, $LDFLAGS, $TEST_LDFLAGS
+if [ -z "${CXXFLAGS+IS_SET}" ]; then
+    CXXFLAGS=""
+    AUTO_CXXFLAGS="true"
+else
+    AUTO_CXXFLAGS="false"
+fi
+if [ -z "${TEST_CXXFLAGS+IS_SET}" ]; then
+    TEST_CXXFLAGS=""
+    AUTO_TEST_CXXFLAGS="true"
+else
+    AUTO_TEST_CXXFLAGS="false"
+fi
+if [ -z "${LDFLAGS+IS_SET}" ]; then
+    LDFLAGS=""
+    AUTO_LDFLAGS="true"
+else
+    AUTO_LDFLAGS="false"
+fi
+if [ -z "${TEST_LDFLAGS+IS_SET}" ]; then
+    TEST_LDFLAGS=""
+    AUTO_TEST_LDFLAGS="true"
+else
+    AUTO_TEST_LDFLAGS="false"
+fi
+
+## Verify PLATFORM value
+if [ "$PLATFORM" != "Linux" ] && [ "$PLATFORM" != "FreeBSD" ] && \
+[ "$PLATFORM" != "OpenBSD" ] && [ "$PLATFORM" != "Darwin" ]; then
+    warning "$PLATFORM" platform is unknown!
+    sub_warning Known Platforms are: Linux, FreeBSD, OpenBSD, Darwin
+fi
+
+## Create testfile.cc to test c++ compiler
+echo "int main(int argc, char **argv) { return 0; }" > testfile.cc || error Cant create test file
+
+## Find and test C++ compiler
+if [ -z "${CXX:-}" ]; then
+    info Checking C++ compiler...
+    for guess in g++ clang++ c++; do
+        findcxx "$guess"
+        if [ -n "${CXX:-}" ]; then
+            sub_info "$CXX"
+            break # Found
+        fi
+    done
+    if [ -z "${CXX:-}" ]; then
+       sub_error No C++ compiler found! # Not found
+    fi
+fi
+cxx_works "$CXX"
+if [ -n "${CXX_FOR_BUILD:-}" ]; then
+    cxx_works "$CXX_FOR_BUILD"
+fi
+
+## Test compiler/linker supported arguments
+if [ "$AUTO_CXXFLAGS" = "true" ]; then
+    for argument in -std=c++11 \
+                    -Wall \
+                    -Os \
+                    -fno-plt
+    do
+        try_cxx_argument CXXFLAGS $argument
+    done
+    if [ "$PLATFORM" != "Darwin" ]; then
+        try_cxx_argument CXXFLAGS -fno-rtti
+    fi
+fi
+if [ "$AUTO_LDFLAGS" = "true" ] && [ "$AUTO_CXXFLAGS" = "true" ]; then
+    try_both_argument CXXFLAGS LDFLAGS -flto
+fi
+if [ "$AUTO_LDFLAGS" = "true" ]; then
+    if [ "$PLATFORM" = "FreeBSD" ]; then
+        try_ld_argument LDFLAGS -lrt
+    fi
+fi
+if [ "${NO_SANITIZE:-}" = "auto" ]; then
+    if [ "$AUTO_TEST_LDFLAGS" = "true" ] && [ "$AUTO_TEST_CXXFLAGS" = "true" ]; then
+        try_both_argument TEST_CXXFLAGS TEST_LDFLAGS -fsanitize=address,undefined
+    elif [ "$AUTO_TEST_LDFLAGS" = "true" ]; then
+        try_ld_argument TEST_LDFLAGS -fsanitize=address,undefined
```

Comment:
Does it really make sense to try this as a link-only argument if not adding it to compilation flags? You would be relying the sanitization flags being specified in the compiler flags by the user, in that case. Same below: does it make sense to add it as a compilation flag if it's not being added in the link flags?

More generally, there's an issue that `--enable-sanitize` doesn't work if TESTCXXFLAGS and TESTLDFLAGS are specified. This needs to be either documented properly or fixed (or, remove the option).

Reply:
So just try `-fsanitize` when both of them are `auto`?

Reply:
See my larger note against line 371. Just trying `-fsanitize` when both `AUTO_TEST_LDFLAGS` and `AUTO_TEST_CXXFLAGS` are true is one option, but then `--enable-sanitize` should do something sensible (eg: either print an error message, or add `-fsanitize` unconditionally and print a message that `TEST_LDFLAGS`/`TEST_CXXFLAGS` were modified if they were specified by the user)